hack/test/e2e-docker.sh

```diff
@@ -29,11 +29,12 @@ function create_cluster {
 }
 
 function destroy_cluster() {
-  "${TALOSCTL}" cluster destroy --name "${CLUSTER_NAME}"
+  "${TALOSCTL}" cluster destroy --name "${CLUSTER_NAME}" --provisioner "${PROVISIONER}"
 }
 
 create_cluster
 get_kubeconfig
 ${KUBECTL} config set-cluster e2e-docker --server https://10.5.0.2:6443
 run_talos_integration_test_docker
 run_kubernetes_integration_test
+destroy_cluster
```

Comment:
I think this is wrong, as I believe we rely on docker cluster to be up in CAPI tests - e2e-docker creates a cluster, e2e-capi installs CAPI, e2e-aws/gcp use that cluster to provision talos in the cloud